examples/src/examples/animation/events.tsx

```diff
@@ -136,7 +136,7 @@ class EventsExample extends Example {
                 let emissive = box.model.material.emissive;
                 emissive = new pc.Vec3(emissive.r, emissive.g, emissive.b);
                 emissive.scale(0.92);
-                emissive = new pc.Color(emissive.data);
+                emissive = new pc.Color(emissive.r, emissive.g, emissive.b);
```

Comment:
Can this whole block handling the emissive color be refactored? It's super-verbose (copy to Vec3 just to multiply by a scalar). Essentially, you just want to lerp colors to black. Might also be a good excuse to add `Color#luminance` to the Color API to avoid the test on each color component?
What's your opinion on all this @mvaligursky?

Reply:
I'd just do this

```
                let emissive = box.model.material.emissive;
                emissive.lerp(emissive, pc.Color.BLACK, 0.3);
                box.model.material.emissive = emissive;

                // // @ts-ignore engine-tsd
                // let emissive = box.model.material.emissive;
                // emissive = new pc.Vec3(emissive.r, emissive.g, emissive.b);
                // emissive.scale(0.92);
                // emissive = new pc.Color(emissive.data);

                // if (emissive.r < 0.001 && emissive.g < 0.001 && emissive.b < 0.001) {
                //     // @ts-ignore engine-tsd
                //     box.model.material.emissive = new pc.Color(0, 0, 0);
                //     popBoxCount++;
                // } else {
                //     // @ts-ignore engine-tsd
                //     box.model.material.emissive = emissive;
                // }

```